### PR TITLE
Save configuration profiles and gesture maps as an atomic operation (…

### PR DESCRIPTION
**Pull request by bramd**
…refs issue #3165)

Instead of directly writing the file, a temporary file is created and
moved to the final location. This prevents corrupt configuration files
being written.
Added a fileUtils module with just a FaultTolerantFile context manager for
now to facilitate this pattern.
Please update miscDependencies submodule to a revision that includes the
osmove module.

### CODE REVIEW

File: source/config/__init__.py [hunk 1]
```diff
@@ -738,7 +742,7 @@ def renameProfile(self, oldName, newName):
 		if oldName.lower() != newName.lower() and os.path.isfile(newFn):
 			raise ValueError("A profile with the same name already exists: %s" % newName)
 
-		os.rename(oldFn, newFn)
+		osreplace.replace(oldFn, newFn)
```
Comment (jcsteh) [inline source/config/__init__.py]: Why the change to osreplace here? In this case, the destination file is not supposed to exist. We should have picked that up above, but it still seems unnecessary... unless I'm missing something?

File: source/fileUtils.py [hunk 2]
```diff
@@ -0,0 +1,16 @@
+import os
```
Comment (jcsteh) [inline source/fileUtils.py]: Copyright header please.

File: source/config/__init__.py (hunk 1 above)
Comment (bramd) [inline source/config/__init__.py]: @jcsteh As I read it, it could be possible this method is called with the same oldName and newName, probably we should check that and just return early instead. Then I agree it's not nescesary as far as I can see.

Comment (bramd) [inline source/config/__init__.py]: Hmm, after looking at the code again, the "same name" case is clearly handled here. SO, I will use os.rename instead.

File: source/fileUtils.py (hunk 2 above)
Comment (bramd) [inline source/fileUtils.py]: @jcsteh Fixed. There are still quite some files without this header in the codebase though. I just didn't think about it and it might have helped if I saw the header every time while browsing parts of the source.

Comment (jcsteh) [inline source/fileUtils.py]: Thank you. Understood. While I don't necessarily want to go through the 
code base just adding/tweaking copyright headers, I'm making an effort 
to fix this for any files I touch.

**Comment by feerrenrut** [comment]
Closing this, as it is now superseded by #6987
